src/Configuration/Configuration.php

```diff
@@ -57,13 +57,23 @@ public static function fromArray(array $args): self
     }
 
     /**
+     * @param  array{
+     *     parameters: array<string, string>,
+     *     formatters: ?array,
+     *     layers: array<array{name: string, collectors: array<array<string, string>>}>,
+     *     paths: list<string>,
+     *     exclude_files: ?array,
+     *     ruleset: array<string, string[]>,
+     *     skip_violations: array<string, string[]>,
+     *     analyser: array<string, mixed>,
+     *     ignore_uncovered_internal_classes: bool,
+     * } $options
+     *
      * @throws InvalidConfigurationException
      */
     private function __construct(array $options)
     {
-        $this->layers = array_map(static function (array $v): ConfigurationLayer {
-            return ConfigurationLayer::fromArray($v);
-        }, $options['layers']);
+        $this->layers = array_map([self::class, 'createLayerConfig'], $options['layers']);
```

Comment:
How about using:
```suggestion
        $this->layers = array_map([ConfigurationLayer::class, 'fromArray'], $options['layers']);
```
That way we don't need to add the new method